--- a/arp.c
+++ b/arp.c
@@ -279,7 +279,7 @@ void rq_process(struct in_addr ipaddr, i
     /* Walk through the list */
     
     while (cur_entry != NULL) {
-	if ( ipaddr.s_addr == *((long *) cur_entry->req_frame.arp.arp_tpa) && ifindex != cur_entry->req_if.sll_ifindex ) {
+	if ( ipaddr.s_addr == ((struct in_addr *) cur_entry->req_frame.arp.arp_tpa)->s_addr && ifindex != cur_entry->req_if.sll_ifindex ) {
 
 	    if (debug)
 	        printf("Found %s in request queue\n", inet_ntoa(ipaddr));
